monitoring/notification/dashboard.py

```diff
@@ -0,0 +1,521 @@
+    legendDisplayMode='table',
+    legendPlacement='right',
+    unit=UNITS.BYTES,
+    targets=[Target(
+        expr='\n'.join([
+            'sum by(target) (',
+            'rate(',
+            '    notification_queue_processor_notification_size[$__rate_interval]',
+            '))',
+        ]),
+        legendFormat='{{target}}',
+    )])
+
+#----------------------------------------------------------------
+
+producer_processing_time = TimeSeries(
+    title='Notification Producer Processing Time',
+    description='Time it takes for the notification producer to process a batch of oplog events',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    lineWidth=3,
+    fillOpacity=10,
+    spanNulls=True,
+    legendDisplayMode='table',
+    legendPlacement='right',
+    unit=UNITS.SECONDS,
+    targets=[Target(
+        expr='\n'.join([
+            'avg(',
+            'rate(',
+            '    queue_populator_processing_delay_sec{origin="notification"}[$__rate_interval]',
+            '))',
+        ]),
+        legendFormat='Notification Producer',
+    )])
+
+processors_processing_time = TimeSeries(
+    title='Notification Processors Processing Time',
+    description='Time it takes for each of the notification processors to process an oplog event',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    lineWidth=3,
+    fillOpacity=10,
+    spanNulls=True,
+    legendDisplayMode='table',
+    legendPlacement='right',
+    unit=UNITS.SECONDS,
+    targets=[Target(
+        expr='\n'.join([
+            'avg by(target) (',
+            'rate(',
+            '    notification_queue_processor_processing_delay_sec[$__rate_interval]',
+            '))',
+        ]),
+        legendFormat='{{target}}',
+    )])
+
+notification_delivery_delay = TimeSeries(
+    title='Delivery Delay',
+    description='Delay between when a notification is sent and when it is received',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    lineWidth=3,
+    fillOpacity=10,
+    spanNulls=True,
+    legendDisplayMode='table',
+    legendPlacement='right',
+    unit=UNITS.SECONDS,
+    targets=[Target(
+        expr='\n'.join([
+            'avg by(target) (',
+            'rate(',
+            '    notification_queue_processor_notification_delivery_delay_sec[$__rate_interval]',
+            '))',
+        ]),
+        legendFormat='{{target}}',
+    )])
+
+#----------------------------------------------------------------
+
+cached_buckets = Stat(
+    title='Cached Notification Configurations',
+    description='Total number of notification configurations cached',
+    dataSource='${DS_PROMETHEUS}',
+    colorMode='value',
+    noValue='0',
+    reduceCalc='last',
+    targets=[Target(
+        expr='\n'.join([
+            'sum by (origin) (',
+            '    notification_config_manager_cached_buckets_count',
```

Comment:
is this one a `counter` or a `gauge` ?
if it is a counter, the expr must use rate() or increase()

Also, i am wondering if we should use a sum() here : we don't really care about the "total", but about the size in each pod. Maybe not aggregating pods would thus be preferable.